astropy/logger.py

```diff
@@ -398,8 +402,8 @@ def log_to_file(self, filename, filter_level=None, filter_origin=None):
             with logger.log_to_file('myfile.log'):
                 # your code here
         '''
-
-        fh = logging.FileHandler(filename)
+        encoding = conf.log_file_encoding if conf.log_file_encoding != "" else None
```

Comment:
I think it would work simply like this?
```suggestion
        encoding = conf.log_file_encoding if conf.log_file_encoding else None
```